sql/src/main/java/org/apache/druid/sql/calcite/parser/DruidSqlParserUtils.java

```diff
@@ -159,7 +160,11 @@ public static Granularity convertSqlNodeToGranularity(SqlNode sqlNode) throws Pa
       catch (IllegalArgumentException e) {
         throw new ParseException(StringUtils.format("%s is an invalid period string", granularitySqlNode.toString()));
       }
-      return new PeriodGranularity(period, null, null);
+      Granularity granularity = new PeriodGranularity(period, null, null);
```

Comment:
This check shouldn't be here, because the method is merely supposed to be converting the given `sqlNode` to a Granularity. This check should be performed either in the Planner or the DruidSqlInsertNode IMO.